pkg/cmd/spks.go

```diff
@@ -55,142 +54,145 @@ func SpksCMD() *cli.Command {
+
+				case <-firstTime:
+					runSPKSBilling(prometheusURL, prometheusQueryArr, logger, startOfToday, allMetrics, err, salesOrder, UnitID, startYesterdayAbsolute, endYesterdayAbsolute, odooClient)
 				}
 			}
-			return nil
 		},
 	}
 }
 
-func QueryPrometheus(ctx context.Context, v1api v1.API, query string, logger logr.Logger, absoluteBeginningTime time.Time) (int, error) {
+func runSPKSBilling(prometheusURL string, prometheusQueryArr [4]string, logger logr.Logger, startOfToday time.Time, allMetrics map[string]map[string]prometheus.Counter, err error, salesOrder string, UnitID string, startYesterdayAbsolute time.Time, endYesterdayAbsolute time.Time, odooClient *odoo.OdooAPIClient) {
+	mariadbStandard, mariadbPremium, redisStandard, redisPremium := getDatabasesCounts(prometheusURL, prometheusQueryArr, logger, startOfToday, allMetrics)
+
+	if mariadbStandard == -1 || mariadbPremium == -1 || redisStandard == -1 || redisPremium == -1 {
+		logger.Error(err, "At least one database query failed, skipping data export to Odoo API")
+	} else {
+		billingRecords := generateBillingRecords(salesOrder, UnitID, startYesterdayAbsolute, endYesterdayAbsolute, mariadbStandard, mariadbPremium, redisStandard, redisPremium)
+
+		err = odooClient.SendData(billingRecords)
+		if err != nil {
+			logger.Error(err, "Error sending data to Odoo API")
+		}
+	}
+}
+
+func generateBillingRecords(salesOrder string, UnitID string, startYesterdayAbsolute time.Time, endYesterdayAbsolute time.Time, mariadbStandard int, mariadbPremium int, redisStandard int, redisPremium int) []odoo.OdooMeteredBillingRecord {
+	billingRecords := []odoo.OdooMeteredBillingRecord{
+		{
+			ProductID:     "appcat-spks-mariadb-standard",
+			InstanceID:    "mariadb-standard",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(mariadbStandard),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+		{
+			ProductID:     "appcat-spks-mariadb-premium",
+			InstanceID:    "mariadb-premium",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(mariadbPremium),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+		{
+			ProductID:     "appcat-spks-redis-standard",
+			InstanceID:    "redis-standard",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(redisStandard),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+		{
+			ProductID:     "appcat-spks-redis-premium",
+			InstanceID:    "redis-premium",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(redisPremium),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
```

Comment:
Small nitpick: Lot's of copy&paste. This can probably be compacted

Reply:
extracted timerange to separate variable, but generation of those provides even more code :D